src/Helpers/Helper.elm

```diff
@@ -99,15 +99,10 @@ assumptions : Zipper.Zipper -> List (Signed Formula)
 assumptions z =
     (++)
         (Maybe.map2 second
-            ((Zipper.zFirstRef z).up
-                |> Maybe.andThen
-                    (\x ->
-                        if x == 0 then
-                            Just ()
-
-                        else
-                            Nothing
-                    )
+            (if List.isEmpty ( Zipper.zNode z |> .references ) then
```

Comment:
It’d be nicer to either simply apply the predicate to the list of references `List.isEmpty (Zipper.zNode z).references`, or use the “pipeline“ throughout `z |> Zipper.zNode |> .references |> List.isEmpty`.